### ISSUE TITLE
cloned_ref_to_slice_refs doesn't trigger on implicit clone

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16320]
### Summary

The `cloned_ref_to_slice_refs` lint does not trigger for implicit clones such as `to_owned()`.

### Lint Name

`cloned_ref_to_slice_refs`

### Reproducer

I tried this code:

```rust
let _a = String::new();
let _b = &[_a.to_owned()];
```

Expected behavior:
The `cloned_ref_to_slice_refs` lint should trigger.

Actual behavior:
The lint does not trigger.

[godbolt.org](https://godbolt.org/z/GhWGx3Tsj)

### Version

```text
nightly
```

**Comment by xtqqczze** [comment]
There are other cases of implicit clone, for example:

```rust
let _a = vec![1, 2, 3];
let _b = &[_a.to_vec()];
```

More generally, anything that would trigger the `implicit_clone` lint:

https://github.com/rust-lang/rust-clippy/blob/86eaeaadd935bfc6451602e8762f4a23305becaf/clippy_lints/src/methods/implicit_clone.rs#L44-L55
